### PR TITLE
*: support encryption context

### PR DESCRIPTION
**Pull request by gyuho**
Use cases:
- Create grant with `GrantConstraints` https://docs.aws.amazon.com/kms/latest/APIReference/API_GrantConstraints.html
- Encrypt with `EncryptionContext` https://docs.aws.amazon.com/kms/latest/APIReference/API_Encrypt.html

e.g.

```sh
./server --encryption-context=a=b,c=d,e=hello-world
```

### CODE REVIEW

File: pkg/plugin/plugin_test.go [hunk 1]
```diff
@@ -32,27 +33,36 @@ var (
...
 		{
 			input:  plainMessage,
+			ctx:    nil,
 			output: encryptedMessage,
 			err:    nil,
 		},
 		{
 			input:  plainMessage,
-			output: encryptedMessage,
+			ctx:    nil,
```
Comment (micahhausler) [inline pkg/plugin/plugin_test.go]: Can you add one test case with a non-nil, but empty map?
